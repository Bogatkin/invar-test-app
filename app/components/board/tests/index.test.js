import React from 'react';
import renderer from 'react-test-renderer';

import BoardRecord from 'models/board';
import BoardItemRecord from 'models/boardItem';
import { List } from 'immutable';

import Board from '../index';

describe('<Board />', () => {
  it('renders correctly', () => {
    const testBoard = new BoardRecord({
      id: 1,
      name: 'Board 1',
      items: new List([
        new BoardItemRecord({ top: 100, left: 200, color: '#ff0000', type: 'box' }),
        new BoardItemRecord({ top: 300, left: 200, color: '#ffff00', type: 'circle' }),
      ]),
    });

    const OriginalBoard = Board.DecoratedComponent;
    const identity = (el) => el;

    // TODO: update test
    const tree = renderer.create(
      <OriginalBoard
        board={testBoard}
        connectDropTarget={identity}
        changeItemPosition={identity}
        removeItem={identity}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
