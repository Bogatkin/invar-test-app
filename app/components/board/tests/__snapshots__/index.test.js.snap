exports[`<Board /> renders correctly 1`] = `
<div
  className="box b-board">
  <div
    className="b-board__item b-board__item--box"
    style={
      Object {
        "background": "#ff0000",
        "left": 200,
        "top": 100,
      }
    } />
  <div
    className="b-board__item b-board__item--circle"
    style={
      Object {
        "background": "#ffff00",
        "left": 200,
        "top": 300,
      }
    } />
</div>
`;
